source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/SqlStatementFactoryTests.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResultClient;
+using FluentAssertions;
+using NodaTime;
+using Xunit;
+using Xunit.Categories;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.UnitTests.Infrastructure.CalculationResultClient;
+
+[UnitTest]
+public class SqlStatementFactoryTests
+{
+    [Fact]
+    public void CreateForSettlementReport_ReturnsExpectedSqlStatement()
+    {
+        // Arrange
+        var gridAreasCodes = new[] { "123", "234", "345" };
+        var periodStart = Instant.FromUtc(2022, 10, 12, 1, 0);
+        var periodEnd = Instant.FromUtc(2022, 10, 12, 3, 0);
+        const string expectedSql = "SELECT grid_area, batch_process_type, time, time_series_type, quantity FROM wholesale_output.result WHERE grid_area IN (123,234,345) AND time_series_type IN ('production','flex_consumption','non_profiled_consumption','net_exchange_per_ga') AND batch_process_type = 'BalanceFixing' AND time BETWEEN '2022-10-12T01:00:00Z' AND '2022-10-12T03:00:00Z' AND aggregation_level = 'total_ga' order by time";
```

Comment:
How about making it much faster to understand/read the test? Perhaps something similar to
```suggestion
    [Theory]
    [InlineAutoMoqData]
    public void CreateForSettlementReport_ReturnsExpectedSqlStatement(List<string> gridAreaCodes)
    {
        // Arrange
        var expectedSql = CreateSettlementReportSql(gridAreaCodes);
```

Reply:
I just like that the expected string is hard coded, and not relying on the conversion of an array to string etc.

Reply:
And I like to be able to very quickly read and understand what's going on and what's the intent of tests and only delve into detail on a need-to basis :-) It's a general pattern I aim to always use when coding "delegate detail".
But feel free to make your choice.

Reply:
I agree with the delegation of  details. However, I think in this case it also have some benefits to keep it as is (sql string is hard coded and placed next to the other variables).